core/slim_sim.cpp

```diff
@@ -8443,12 +8300,140 @@ void SLiMSim::__ConfigureSubpopulationsFromTables(EidosInterpreter *p_interprete
 		}
 		
 		char *metadata_char = pop_table.metadata + pop_table.metadata_offset[pop_index];
+		std::string metadata_string(metadata_char, metadata_length);
+		nlohmann::json subpop_metadata;
 		
-		if (metadata_length < sizeof(SubpopulationMetadataRec))
-			EIDOS_TERMINATION << "ERROR (SLiMSim::__ConfigureSubpopulationsFromTables): malformed population metadata; this file cannot be read." << EidosTerminate();
+		try {
+			subpop_metadata = nlohmann::json::parse(metadata_string);
+		} catch (...) {
+			EIDOS_TERMINATION << "ERROR (SLiMSim::__ConfigureSubpopulationsFromTables): population metadata does not parse as a valid JSON string; this file cannot be read." << EidosTerminate(nullptr);
+		}
 		
-		SubpopulationMetadataRec *metadata = (SubpopulationMetadataRec *)metadata_char;
-		slim_objectid_t subpop_id = metadata->subpopulation_id_;
+		if (!subpop_metadata.is_object())
+			EIDOS_TERMINATION << "ERROR (SLiMSim::__ConfigureSubpopulationsFromTables): population metadata does not parse as a JSON object; this file cannot be read." << EidosTerminate(nullptr);
+		
+		slim_objectid_t subpop_id;
+		
+		if (!subpop_metadata.contains("slim_id"))
+			EIDOS_TERMINATION << "ERROR (SLiMSim::__ConfigureSubpopulationsFromTables): required population metadata key 'slim_id' is missing; this file cannot be read." << EidosTerminate(nullptr);
```

Comment:
So I guess to make `slim_id` optional you'd have to at this point skip this entry. I'm not sure how annoying that would be.

Reply:
See comment previous.